internal/db/migrations/2024-08-28.0-tss_transactions.sql

```diff
@@ -0,0 +1,37 @@
+-- +migrate Up
+
+CREATE TABLE tss_transactions (
+    transaction_hash VARCHAR(70) PRIMARY KEY,
+    transaction_xdr VARCHAR(400),
+    webhook_url VARCHAR(250),
```

Comment:
Weren't we going to store the webhook URL on the application level since we expect it would generally be the same for all transactions?

Reply:
True, I know that at first we figured they'd be a 1:1 correspondence between client and TSS, but I thought about this and I do think it is possible that the client may want to have more than one webhook. For example, they may only have one instance of TSS/wallet backend running for multiple of their applications, each one of which will have a different webhook. They may also want to have different webhooks for testing vs production

Reply:
> I do think it is possible that the client may want to have more than one webhook

Yeah I think that is possible indeed. How would the clients specify the webhook URL in this case (that would feed this column in the DB)?

> They may also want to have different webhooks for testing vs production

That I don't think would be possible because the wallet backend is very environment specific (e.g.: your distribution account will only exist in a single network, the RPC will only be connected to a single network, etc.)

Reply:
There's 2 ways I can think of in which the clients can specify their webhook url:

1. We have an explicit endpoint that they can call to register their endpoint
2. They send their webhook endpoint along with their transactions

I like 2 better, but open to other ideas

Reply:
The downsides I see with 2 is that it results in a lot of duplicate data in the `tss_transactions` table and makes the process of submitting transactions a little less simple.
To me this feels like project level type of setting that could be set once and the client wouldn't have to worry about it anymore. Given that I think I'd lean to the 1. side. Wdyt?

Reply:
If we go with 1. I picture it looking something like this:

1. we will have a /register-webhook endpoint for the clients to call where they provide a client_id (string) and a webhook url (string)
2. we will have a webhooks table that will store the client id along with the webhook url
3. Now with every transaction submission request, the client will send their client-id instead of the webhook url
4. So we will have to add a client id column to the transactions table so we know which client to send the transaction result to

Alternatively, we could also generate the client id ourselves (could just an auto incrementing id) when the client calls the /register-webhook endpoint and just have the client give us that client id for transaction submission requests

Thing is, we will still have that client id column in the transactions table which will have a lot of duplicate data

Thoughts?

Reply:
I like the idea of us generating the client id on the /register-webhook endpoint better

Reply:
Our slack discussion summarized below:

Decides to keep the table design as is because it was decided that a client can have more than 1 webhook url. We will have the client pass in the webhook url with every transaction submission request